Zero-shot image classification task guide (#22132)
huggingface/transformers: docs/source/en/tasks/zero_shot_image_classification.mdx

<!--Copyright 2023 The HuggingFace Team. All rights reserved.

Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
the License. You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
specific language governing permissions and limitations under the License.
-->

# Zero-shot image classification

[[open-in-colab]]

Zero-shot image classification is a task that involves classifying images into different categories using a model that was
not explicitly trained on data containing labeled examples from those specific categories.

Traditionally, image classification requires training a model on a specific set of labeled images, and this model learns to
"map" certain image features to labels. When there's a need to use such model for a classification task that introduces a
new set of labels, fine-tuning is required to "recalibrate" the model.

> In contrast, a pre-trained model that can be used for zero-shot image classification typically has been trained on a large
> dataset of images and associated descriptions. Such model has learned aligned vision-language representations. These can
> be used for many downstream tasks including zero-shot image classification.

Reviewer: Just rephrasing to clarify the use of multi-modal models for zero-shot image :
```suggestion
In contrast, zero-shot or open vocabulary image classification models are typically multi-modal models that have been trained on a large
dataset of images and associated descriptions. These models learn aligned vision-language representations that can be used for many downstream tasks including zero-shot image classification.
```

This is a more flexible approach to image classification that allows models to generalize to new and unseen categories

> without the need for additional training data.

Reviewer: I think it'd be great to emphasize the use of free-form text queries as opposed to rigid labels:
```suggestion
without the need for additional training data and enables users to query images with free-form text descriptions of their target objects .
```
